src/client/handlers/AuthSignToken.ts

```diff
@@ -0,0 +1,63 @@
+import type {
+  ClientRPCRequestParams,
+  ClientRPCResponseResult,
+  IdentityRequestData,
+  IdentityResponseData,
+  TokenIdentityRequest,
+  TokenIdentityResponse,
+} from '../types.js';
+import type KeyRing from '../../keys/KeyRing.js';
+import type { PublicKey } from '../../keys/types.js';
+import { UnaryHandler } from '@matrixai/rpc';
+import Token from '../../tokens/Token.js';
+import * as clientErrors from '../errors.js';
+import * as nodesUtils from '../../nodes/utils.js';
+import * as tokensUtils from '../../tokens/utils.js';
+
+class AuthSignToken extends UnaryHandler<
+  {
+    keyRing: KeyRing;
+  },
+  ClientRPCRequestParams<TokenIdentityRequest>,
+  ClientRPCResponseResult<TokenIdentityResponse>
+> {
+  public handle = async (
+    input: ClientRPCRequestParams<TokenIdentityRequest>,
+  ): Promise<TokenIdentityResponse> => {
+    const { keyRing }: { keyRing: KeyRing } = this.container;
+
+    // Get and verify incoming node
+    const inputToken = { payload: input.payload, signatures: input.signatures };
+    const incomingToken = Token.fromEncoded<TokenIdentityRequest>(inputToken);
+    const inputPayload = tokensUtils.parseTokenPayload<IdentityRequestData>(
+      inputToken.payload,
+    );
```

Comment:
`Token.fromEncoded` does all of the parsing for you. There's no need to parse the payload again.